tiledb/test/src/util.rs

```diff
@@ -0,0 +1,9 @@
+use proptest::prelude::*;
+
+pub fn prop_field_name() -> impl Strategy<Value = String> {
+    proptest::string::string_regex("[a-zA-Z0-9_]*")
+        .expect("Error creating field name strategy")
+        .prop_filter("Field names may not begin with reserved prefix", |name| {
```

Comment:
Is this generally true, or just true for attributes?

Reply:
Pretty sure it applies to domain names as well but I'll double check to make sure.

Reply:
I should also mention, `field_name` is the naming convention used in core to mean "attribute or dimension name" in case that wasn't obvious.

Reply:
Huh. Looks like not actually. Although `__coords` has a special meaning but in a way that we don't want to exclude it. I'll swap that back to a prop_attribute_name and prop_dimension_name.